compiler/parser.nim

```diff
@@ -1500,7 +1504,7 @@ proc postExprBlocks(p: var Parser, x: PNode): PNode =
     result = makeCall(result)
     getTok(p)
     skipComment(p, result)
-    if p.tok.tokType notin {tkOf, tkElif, tkElse, tkExcept, tkFinally}:
+    if not (p.tok.tokType in {tkOf, tkElif, tkElse, tkExcept, tkFinally} and sameInd(p)):
```

Comment:
Doesn't this imply that now

```nim

match x
of ...

```

does not work anymore? Or maybe

```nim

match x
  of ...

```

doesn't anymore?

Reply:
This is not the actual parsing step, this is just the lookahead triggered by the colon. The actual parsing below does check for `sameInd` (the `while sameInd(p)` part), so only the following worked, which is still true:

```nim
match x:
of ...
```

The problem is that this lookahead ate the colon token when any `of/else/...` token followed, regardless of indent.

```nim
  match x: # colon here gets eaten because the next token is `else`
else: ... # but not same indent, so the call just becomes match(x)
```

This PR just corrects the lookahead to trigger only when post expr blocks will be parsed.